### DISCUSSION TITLE
Streaming the file problem

### DISCUSSION THREAD

**Discussion by moondayyy** [microsoft/msquic#3805] [Q&A] [answered]
I want to send a file from the client to the server with the StreamSend method, but the file is sent once on the client side, but it is automatically divided into parts on the server side. 

If I know the size of the file I want to send on the server side, I can combine it with **Event->RECEIVE.AbsoluteOffset**. 
What does the Context parameter in my **ServerStreamCallback(HQUIC Stream, void \*Context, QUIC_STREAM_EVENT \*Event)** function do on the server side, can I send this parameter with the Context parameter of the client-side **StreamSend(stream, &buffer, 1, streamSendFlag,Context)** function?

**Comment by nibanks** [comment, marked as answer]
The `Context` pointer is just a local (to your process) object that is passed back to you from when you originally registered your callback, either via from [StreamOpen](https://github.com/microsoft/msquic/blob/main/docs/api/StreamOpen.md) (if you created the stream) or when you called [SetCallbackHandler](https://github.com/microsoft/msquic/blob/main/docs/api/SetCallbackHandler.md) (if the peer created it). It's not something that will be usable by the peer.

If you need this information to be given to the peer then you will need to encode it on the wire and send it too, perhaps as some kind of metadata header for your stream data. For instance, if all you want to send is the total size, encode that in the first few bytes of the stream explicitly.
